instructor/common.py

```diff
@@ -68,19 +68,33 @@ def create_new_assignment(request):
 
 
 def assignment_setup(request):
+    error = ''
     if request.method == "POST":
-        # Have to give error for empty name
-        request.session['assignment_name'] = request.POST['name']
+        assignment_name = request.POST['name']
         if request.POST['based_on']:
             request.session['based_on'] = request.POST['based_on']
-
-        if 'continue' in request.POST:
-            return redirect('common_course_setup')
+        if assignment_name:
+            request.session['assignment_name'] = assignment_name
+            if 'continue' in request.POST:
+                return redirect('common_course_setup')
+        else:  # Have to give error for empty name
+            error = "*Empty assignment name"
+    else:
+        index = 1
+        found = False
+        while not found:
+            assignment_name = "Assignment" + str(index)
+            if models.Assignment.objects.filter(name=assignment_name).count() > 0:
+                index += 1
+            else:
+                found = True
 
     all_assignments = models.Assignment.objects.all()
+
     return render_to_response('instructor/assignment_setup.html',
                               {'assignments': all_assignments,
-                               'assignment_name': request.session['assignment_name'],
+                               'error': error,
+                               'assignment_name': assignment_name,
                                'based_on': request.session['based_on'],
                                'new': request.session['new']},
                                 context_instance=RequestContext(request))
```

Comment:
not sure if there is any linting or pep8 check, but might want to re-indent this anyway for readability